portal/templates/portal_wrapper.html

```diff
@@ -170,37 +139,33 @@
-      <div class="modal-body">{{ _("Your session will expire in approximately") }} <span id="remaining-time"></span> {{ _("seconds due to inactivity.") }}</div>
-      <div class="modal-footer"><button id="log-out" class="btn btn-default" type="button" data-dismiss="modal">Log Out</button><button id="stay-logged-in" class="btn btn-warning" type="button" data-dismiss="modal">{{ _("Stay Logged In") }}</button></div>
+      <div class="modal-header"><h4 class="modal-title" id="sessWarnLabel">{{_("Your session is about to expire")}}</h4></div>
+      <div class="modal-body">
+        <span id="remaining-time">{{_("Your session will expire in approximately {time} seconds due to inactivity.")}}</span></div>
+      <div class="modal-footer"><button id="log-out" class="btn btn-default" type="button" data-dismiss="modal">{{_("Log Out")}}</button><button id="stay-logged-in" class="btn btn-warning" type="button" data-dismiss="modal">{{ _("Stay Logged In") }}</button></div>
     </div><!-- /.modal-content -->
   </div><!-- /.modal-dialog -->
 </div><!-- /.modal -->
-<script>
-$(document).ready(function(){
-  // Configure and start the session timeout monitor
-  {% if user %}
-      var sessMon = sessionMonitor({
-            sessionLifetime: DEFAULT_SESSION_LIFETIME,
-            timeBeforeWarning: 1 * 60 * 1000,
-            minPingInterval: 1 * 60 * 1000,  // 1 minute
-            activityEvents: "mouseup",
-            pingUrl: '{{PORTAL}}/api/ping',
-            logoutUrl: '{{PORTAL}}/logout',
-            timeoutUrl: '{{PORTAL}}/logout?timed_out=1',
-            modalShown: false,
-            intervalMonitor: false,
-            onwarning: function() {$("#session-warning-modal").modal("show"); if (sessMon.modalShown) sessMon.intervalMonitor = setInterval(function(){ sessMon.ontimeout() }, 2 * 60 * 1000);}
-      });
-      window.sessMon = sessMon;
-       // Configure the session timeout warning modal
-      $("#session-warning-modal").modal({"backdrop": false,"keyboard": false,"show": false})
-              .on("show.bs.modal", function() { sessMon.modalShown = true})
-              .on("hide.bs.modal", function() { sessMon.modalShown = false; if (sessMon.intervalMonitor) clearInterval(sessMon.intervalMonitor); })
-              .on("click", "#stay-logged-in", sessMon.extendsess)
-              .on("click", "#log-out", sessMon.logout)
-              .find("#remaining-time").text(sessMon.timeBeforeWarning / 1000);
-   {% endif %}
-});
-</script>
 <script defer>setTimeout("$('.no-fouc').removeClass('no-fouc');", 300);</script>
+<!-- session monitor script -->
+{% if user %}
+<script type="text/javascript">
+    (function() {
+        var d=document, g=d.createElement("script"), s=d.getElementsByTagName("head")[0];
+        g.type="text/javascript"; g.async=true; g.defer=true; g.src='{{PORTAL}}{{ url_for("static", filename="js/sessionMonitor.js") }}'; 
```

Comment:
Are you able to statically include `sessionMonitor.js` as an external script (outside the `<head>` element)?

Reply:
@ivan-c ivan, I am hesitant to make the change here because i think that was my initial approach but failed miserably - it seems browser automatically stripped the script tag referencing external script on the intervention side - at least that was my observation from CRPO side.  I still cannot figure out why - the approach to get portal wrapper on the CPRO side didn't involve load() method at all - it was straightforward ajax call (then populating wrapper content using .html call on the target element).    I am taking the approach to add script dynamically here because this is how we are including the piwik external script file (i.e. piwik.cirg.washington.edu/piwik.php) in the portal wrapper and that seems to be working.

Reply:
>automatically stripped the script tag referencing external script on the intervention side - at least that was my observation from CRPO side

Interesting, maybe something to do with CORS, or Mark wasn't able to get load() to work?


We should eventually move the piwik variables out of the inline script to an HTML element, or sessionStorage might be better.

In any case, it's still a positive change to pull the sessionMonitor code out of the wrapper, I think the browser should still be able to cache it, even if you're adding it dynamically.